server/src/main/java/com/hedera/block/server/persistence/StreamPersistenceHandlerImpl.java

```diff
@@ -118,12 +118,14 @@ public void onEvent(ObjectEvent<SubscribeStreamResponse> event, long l, boolean
                             metricsService.get(StreamPersistenceHandlerError).increment();
                             throw new BlockStreamProtocolException(message);
                         } else {
-                            // Persist the BlockItem
-                            Optional<BlockItem> result = blockWriter.write(blockItem);
+                            // Persist the BlockItems
+                            List<BlockItem> blockItems =
+                                    subscribeStreamResponse.blockItems().blockItems();
+                            Optional<List<BlockItem>> result = blockWriter.write(blockItems);
```

Comment:
We will have to figure out what Optional to return here so the downstream response can be generated. This is probably ok for now.

Reply:
should we add it to an epic? Maybe the Persist Story Epic https://github.com/hashgraph/hedera-block-node/issues/194

Reply:
I added this item: "
Review Optional return value from the Writer so downstream code can generate the correct response"